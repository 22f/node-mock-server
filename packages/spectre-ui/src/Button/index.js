// @flow

import "spectre.css";
import React from "react";

import noop from "../utils/noop.js";
import type { ButtonClassPropsType, ButtonPropsType } from "../../spectre-ui.js.flow";

function getClassName({ modifier, size, className }: ButtonClassPropsType): string {
	const classNames: Array<string> = ["btn"];

	if (modifier) {
		classNames.push(`btn-${modifier}`);
	}

	if (size) {
		classNames.push(`btn-${size}`);
	}

	if (className) {
		classNames.push(className);
	}

	return classNames.join(" ");
}

const Button = ({
	modifier,
	size,
	className,
	type = "button",
	disabled = false,
	tabIndex = -1,
	label,
	onClick = noop
}: ButtonPropsType): React$Element<*> => (
	<button
		className={getClassName({ modifier, size, className })}
		type={type}
		disabled={disabled}
		tabIndex={tabIndex}
		onClick={onClick}
	>
		{label}
	</button>
);

export default Button;
