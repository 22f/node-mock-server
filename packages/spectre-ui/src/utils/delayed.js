// @flow

const delayed = (fn: Function, delay?: number = 1): string => {
	const timerId: string = window.setTimeout((): Function => {
		window.clearTimeout(timerId);
		return fn.call();
	}, delay);

	return timerId;
};

export default delayed;
