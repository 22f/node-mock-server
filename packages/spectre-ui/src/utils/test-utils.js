/* global describe it expect */
import React from "react";
import { shallow } from "enzyme";

export function itWithoutProp(Component, props, propName: string) {
	it(`snapshot: no ${propName}`, () => {
		const _props = { ...props, [propName]: undefined };
		const tree = shallow(<Component {..._props} />);
		expect(tree).toMatchSnapshot();
	});
}
export function itOptionalProps(Component, props, propNames: Array<string>) {
	propNames.forEach((propName: string) => {
		itWithoutProp(Component, props, propName);
	});
}
export function itEmptyArrayProp(Component, props, propName: string) {
	it(`snapshot: empty ${propName}`, () => {
		const _props = { ...props, [propName]: [] };
		const tree = shallow(<Component {..._props} />);
		expect(tree).toMatchSnapshot();
	});
}
export function itArrayProps(Component, props, propNames: Array<string>) {
	propNames.forEach((propName: string) => {
		itEmptyArrayProp(Component, props, propName);
	});
}
