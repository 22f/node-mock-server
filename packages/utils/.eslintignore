/.idea/
/node_modules/
/dist/
/coverage/
/flow-typed/
/node-mock-server-utils.js.flow
