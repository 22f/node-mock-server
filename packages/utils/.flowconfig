[ignore]
<PROJECT_ROOT>/node_modules/.*.json
<PROJECT_ROOT>/dist/.*
.*/log/.*

[include]

[libs]
./node_modules/node-mock-server-log/node-mock-server-log.js.flow

[options]
unsafe.enable_getters_and_setters=true
esproposal.decorators=ignore
