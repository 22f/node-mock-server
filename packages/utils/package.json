{
    "name": "node-mock-server-utils",
    "version": "0.1.0",
    "private": false,
    "dependencies": {
        "@sindresorhus/is": "0.3.0",
        "camel-case": "3.0.0",
        "capitalize": "1.0.0",
        "del": "3.0.0",
        "make-dir": "1.0.0",
        "node-mock-server-log": "0.1.0",
        "to-pascal-case": "1.0.0"
    },
    "devDependencies": {
        "eslint-config": "0.1.0",
        "flow-bin": "0.61.0",
        "flow-typed": "2.2.3",
        "node-flow-config": "0.1.0"
    },
    "main": "dist/index.js",
    "scripts": {
        "dev": "node-flow-config start",
        "build": "node-flow-config build",
        "test": "CI=true node-flow-config test --env=jsdom --coverage",
        "test:dev": "node-flow-config test --env=jsdom",
        "lint": "npm-run-all lint:*",
        "lint:js": "node_modules/.bin/eslint .",
        "flow": "node_modules/.bin/flow",
        "flow-restart": "node_modules/.bin/flow stop && node_modules/.bin/flow",
        "flow-install": "node_modules/.bin/flow-typed install -i dev"
    }
}
