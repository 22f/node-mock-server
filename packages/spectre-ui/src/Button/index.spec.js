/* global describe it expect */

import React from "react";
import { shallow } from "enzyme";
import { itOptionalProps } from "../utils/test-utils";
import Button from "./index";

const props: Object = {
	disabled: false,
	tabIndex: -1,
	type: "button",
	label: "label",
	modifier: "primary",
	size: "sm",
	className: "className",
	onClick() {}
};

describe("Button", () => {
	describe("render", () => {
		it("snapshot: default", () => {
			const _props = { ...props };
			const tree = shallow(<Button {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		itOptionalProps(Button, props, [
			"disabled",
			"tabIndex",
			"type",
			"label",
			"modifier",
			"size",
			"className",
			"onClick"
		]);
		it("snapshot: disabled", () => {
			const _props = { ...props, disabled: false };
			const tree = shallow(<Button {..._props} />);
			expect(tree).toMatchSnapshot();
		});
	});
});
