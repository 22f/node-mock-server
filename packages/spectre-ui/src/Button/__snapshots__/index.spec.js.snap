// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render snapshot: default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: disabled 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 19,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 10,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 20,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no className 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className={undefined}
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 15,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": undefined,
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 16,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className={undefined}
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no disabled 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 4,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no label 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
/>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label={undefined}
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": undefined,
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
/>,
        "_debugID": 10,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label={undefined}
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no modifier 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier={undefined}
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": undefined,
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 12,
        "_renderedOutput": <button
          className="btn btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier={undefined}
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no onClick 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 17,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 9,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 18,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no size 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size={undefined}
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": undefined,
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 7,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 14,
        "_renderedOutput": <button
          className="btn btn-primary className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size={undefined}
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no tabIndex 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 6,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;

exports[`Button render snapshot: no type 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-sm className"
    disabled={false}
    onClick={[Function]}
    tabIndex={-1}
    type="button"
>
    label
</button>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-sm className"
      disabled={false}
      onClick={[Function]}
      tabIndex={-1}
      type="button"
>
      label
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        className="className"
        disabled={false}
        label="label"
        modifier="primary"
        onClick={[Function]}
        size="sm"
        tabIndex={-1}
        type="button"
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "className",
          "disabled": false,
          "label": "label",
          "modifier": "primary",
          "onClick": [Function],
          "size": "sm",
          "tabIndex": -1,
          "type": "button",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
        "_debugID": 8,
        "_renderedOutput": <button
          className="btn btn-primary btn-sm className"
          disabled={false}
          onClick={[Function]}
          tabIndex={-1}
          type="button"
>
          label
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    className="className"
    disabled={false}
    label="label"
    modifier="primary"
    onClick={[Function]}
    size="sm"
    tabIndex={-1}
    type="button"
/>,
}
`;
